backend/src/main/java/com/huellapositiva/domain/service/EmailCommunicationService.java

```diff
@@ -69,4 +69,26 @@ public void sendSubmittedProposalRevision(ProposalRevisionEmail proposalRevision
         Email email = Email.createFrom(proposalRevisionEmail, emailTemplate, from);
         emailService.sendEmail(email);
     }
+
+
+
+    public void sendRecoveryPasswordEmail(EmailRecoveryPassword emailRecoveryPassword) {
+        try {
+            EmailTemplate emailTemplate = templateService.getRecoveryEmailTemplate(emailRecoveryPassword.getHash());
+            Email email = Email.createFrom(emailRecoveryPassword, emailTemplate, from);
+            emailService.sendEmail(email);
+        } catch (RuntimeException ex) {
+            log.error("Failed to send email:", ex);
+        }
+    }
+
+    public void sendConfirmationPasswordChanged(EmailAddress emailAddress) {
+        try {
+            EmailTemplate emailTemplate = templateService.getConfirmationPasswordChangedTemplate();
+            Email email = Email.createFrom(emailAddress, emailTemplate, from);
+            emailService.sendEmail(email);
+        } catch (RuntimeException ex) {
+            log.error("Failed to send email:", ex);
```

Comment:
Do not throw or catch RuntimeExceptions unless necessary. Create more concrete exception for your use case.

If we are not able to send the email, shouldn't we respond with 500? Otherwise the user will think the process went fine and will expect to get an email in its inbox. Which will not finally arrive and will make it get pissed.

Reply:
Create API tests for this case please if you haven't already. It is an unhappy but necessary path

Reply:
En el mensaje de arriba pusimos eso, que lo veremos mejor en el refactor general en otra rama que no sea esta porque excede el propósito de esta rama.